gatsby-node.js

```diff
@@ -1,7 +1,61 @@
-/**
- * Implement Gatsby's Node APIs in this file.
- *
- * See: https://www.gatsbyjs.org/docs/node-apis/
- */
+const path = require('path');
 
-// You can delete this file if you're not using it
+const CONTENT_DIR = 'src/content/';
+const TEMPLATE_DIR = 'src/templates/';
+
+exports.onCreateNode = ({ node, getNode, actions }) => {
+  if (node.internal.type === 'Mdx') {
+    const { createNodeField } = actions;
+    const { fileAbsolutePath } = node;
+
+    // Get the file path to be used as the page location
+    const value = fileAbsolutePath
+      .replace(`${process.cwd()}/${CONTENT_DIR}`, '')
+      .replace('.mdx', '');
```

Comment:
Do we want to use the `createFilePath` method from `gatsby-source-filesystem` plugin to generate the relative path instead?
```suggestion
const fileRelativePath = createFilePath({ node, getNode, trailingSlash: false });
```